feat: add backend and frontend packages for iValt biometrics

KingOfTac opened the pull request:
<!---
Thanks for filing a pull request! Before you submit, please read the following:

Search open/closed issues before submitting. Someone may have pushed the same thing before!

Provide a summary of your changes in the title field above.
-->

# Pull Request

## Description

<!---
Provide some background and a description of your work.
What problem does this change solve?
Is this a breaking change, chore, fix, feature, etc?
-->
This PR adds two packages:
1. `@trans-stat/node-mfa` contains a simple backend wrapper around iValt's MFA API. The implementation is a simple javascript class that can be instantiated with an iValt API key. The implementation tries to be a simple as possible so that it is easy to extend and integrate with any node.js framework.
2. `@trans-stat/auth-components` is a library of FAST components for easy drop in user authentication UX patterns. Right now this library contains a single `<mfa-card>` component that is intended to be used with the node-mfa backend package. This library will be expanded overtime with more UX patterns related to user authentication.


### Issues

<!---
* List and link relevant issues here.
-->

## Reviewer Notes

<!---
Provide some notes for reviewers to help them provide targeted feedback and testing.

Do you recommend a smoke test for this PR? What steps should be followed?
Are there particular areas of the code the reviewer should focus on?
-->

## Test Plan

<!---
Please provide a summary of the tests affected by this work and any unique strategies employed in testing the features/fixes.
-->

## Checklist

### General

<!--- Review the list and put an x in the boxes that apply. -->

- [ ] I have added tests for my changes.
- [X] I have tested my changes.
- [X] I have updated the project documentation to reflect my changes.

## Next Steps

<!---
If there is relevant follow-up work to this PR, please list any existing issues or provide brief descriptions of what you would like to do next.
-->
Next steps are:
1. ~~Create a MFA backend package for adding traditional 2FA and MFA capabilities to applications~~ This will be added to the node-mfa package
2. Add stackblitz examples demonstrating setting up the backend package with various backend frameworks
3. Add stackblitz examples demonstrating advanced usage of the frontend components and how to customize the Web Components.
4. ~~Add documentation on how to enable vscode intellisense for the Web Components in html and css files.~~ done

scomea commented on packages/web-components/src/biometric-card/biometric-card.ts:
```diff
@@ -0,0 +1,63 @@
+    const data = formValues(this.form) as BiometricCardFormData;
+
+    this.callback?.(data);
+    this.$emit('biometric-request', data);
```
What are your thoughts on event bubbling or not?  I've been tending to turn bubbling off as a micro-optimization in cases where I figure devs will tend to hook listeners directly to the component anyway as it is safe to turn it on later if there is a need.

KingOfTac replied:
What's the optimization benefit to turning it off? I'll admit I haven't dug too deep into configuring events beyond passing the name and event detail, so any help here would be appreciated.

scomea replied:
I assume that there is more work going on in the background to handle a bubbling event vs. a non-bubbling, but I haven't tested to see what impact it actually has.  To control bubbling you pass it as an option `this.$emit("someEvent", eventDetail, {bubbles: false});`

KingOfTac commented:
@bentomands added you as a reviewer to double check formatting and information in the documentation. Here's a list of the files to check:
- packages/biometrics/README.md
- packages/web-components/README.md

awentzel commented on .gitignore:
```diff
@@ -0,0 +1,130 @@
+# Logs
```
Can you remove all the ignore lines that are irrelevant to our technology stack, i.e.; `.grunt` for example?

KingOfTac replied:
We could but it doesn't hurt anything. This is just the standard node.js gitignore that GitHub initializes repositories with.

awentzel commented on .vscode/launch.json:
```diff
@@ -0,0 +1,12 @@
+{
+  "version": "0.2.0",
+  "configurations": [
+    {
```
what happens if user does not have edge?

KingOfTac replied:
I'm actually removing this file since I am going a different route for local debugging and running.

awentzel commented on README.md:
```diff
@@ -9,3 +9,6 @@ Protection provided by [QuSecure](https://www.qusecure.com)
 Authentication provided by [iValt](https://www.ivalt.com)
 
+## Packages
+- [`@trans-stat/biometrics`](https://github.com/trans-stat/tas-security/blob/main/packages/biometrics) - Node.js esm and common.js compatible wrapper around iValt's biometric authentication API
```
should `esm` be all caps?

awentzel commented on beachball.config.js:
```diff
@@ -0,0 +1,15 @@
+module.exports = {
+	disallowedChangeTypes: ['major'],
+  access: 'public',
```
formatting is off? Are we using prettier or something equivalent on merge to master things stay in desired format of the file type?

KingOfTac replied:
Not at this time, no. Prettier is pretty heavy and very opinionated so I have been looking for alternatives that are smaller, faster, and easier to configure.

awentzel commented on packages/web-components/README.md:
````diff
@@ -0,0 +1,87 @@
+}
+```
+
+### `<biometric-card>`
````
iValt doesn't just provide biometrics, so I wonder if the naming is accurate here? For example, goefencing could be one factor of the MFA, which is not biometrics or even date time values.

KingOfTac replied:
I'm open to suggestions on naming. We could even just call this mfa-card and reuse it for other forms of multifactor.

awentzel commented on packages/web-components/src/biometric-card/define.ts:
```diff
@@ -0,0 +1,46 @@
+ * @part submit-button__control - The submit button's internal control element.
+ */
+@customElement({
+  name: 'biometric-card',
```
same question on naming? thoughts?

KingOfTac commented:
@awentzel I renamed the backend package to be more generic so that other MFA service and API wrappers can live there in addition to the iValt wrapper. I also renamed the frontend component to be more generic and am planning on more changes to the component's API in the near future that will enable it to be used for a wider variety of scenarios.
